Modules/_datetimemodule.c

```diff
@@ -130,13 +131,30 @@ class datetime.IsoCalendarDate "PyDateTime_IsoCalendarDate *" "&PyDateTime_IsoCa
 #define MONTH_IS_SANE(M) ((unsigned int)(M) - 1 < 12)
 
 /* Forward declarations. */
-static PyTypeObject PyDateTime_DateType;
-static PyTypeObject PyDateTime_DateTimeType;
-static PyTypeObject PyDateTime_DeltaType;
-static PyTypeObject PyDateTime_IsoCalendarDateType;
-static PyTypeObject PyDateTime_TimeType;
-static PyTypeObject PyDateTime_TZInfoType;
-static PyTypeObject PyDateTime_TimeZoneType;
+typedef struct {
+    PyTypeObject *PyDateTime_DateType;
+    PyTypeObject *PyDateTime_DateTimeType;
+    PyTypeObject *PyDateTime_DeltaType;
+    PyTypeObject *PyDateTime_IsoCalendarDateType;
+    PyTypeObject *PyDateTime_TimeType;
+    PyTypeObject *PyDateTime_TZInfoType;
+    PyTypeObject *PyDateTime_TimeZoneType;
+} _datetimemodule_state;
+
+static inline _datetimemodule_state*
+get_datetime_state(PyObject *module)
+{
+    void *state = PyModule_GetState(module);
+    assert(state != NULL);
+    return (_datetimemodule_state*)state;
+}
+
+static struct PyModuleDef datetimemodule;
+
+#define find_datetimemodule_state_by_type(tp) \
```

Comment:
why define two macros in here?

Reply:
find_datetimemodule_state_by_type could get module state from subclass. It's necessary for us.